### DISCUSSION TITLE
Admin Functions Not Working

### DISCUSSION THREAD

**Discussion by Shawn-E** [supabase/supabase#11666] [Questions]
I am trying to build an app where I need to use admin function to update a users data. Everytime a use the `supabase.auth.admin.getUserById` function with the users id as a parameter in the function. It returns `undefined`. I have already tried to change the `supabaseKey` to the `service_role` when initilizing the client. How can I get these admin function to work?

**Comment by GaryAustin1** [comment]
Make sure you are not also having a signed in client.  It will override the service key used in the client init in the Authorization header.  You might try just setting up a 2nd client depending on what you are doing.

Note this should only be done server side, not on a browser.
